### PR TITLE
add: tls sample

### PR DESCRIPTION
**Pull request by cjphaha**

### CODE REVIEW

File: tls/go-client/app/client.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
+package main
```
Comment (beiwei30) [inline tls/go-client/app/client.go:1]: pls. add apache license header

File: tls/go-client/app/user.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package main
```
Comment (beiwei30) [inline tls/go-client/app/user.go:1]: pls. add apache license.

File: tls/go-client/app/env.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export CONF_CONSUMER_FILE_PATH=../profiles/dev/client.yml
```
Comment (beiwei30) [inline tls/go-client/app/env.sh:1]: avoid use shell script to start. try if build/makefile work or not. pls. refer to HOWTO.md

File: tls/go-client/app/start.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export CONF_CONSUMER_FILE_PATH="../conf/client.yml"
```
Comment (beiwei30) [inline tls/go-client/app/start.sh:1]: avoid use shell script to start. try if build/makefile work or not. pls. refer to HOWTO.md

File: tls/go-server/app/env.sh [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export CONF_PROVIDER_FILE_PATH="../profiles/dev/server.yml"
```
Comment (beiwei30) [inline tls/go-server/app/env.sh:1]: avoid use shell script to start. try if build/makefile work or not. pls. refer to HOWTO.md

File: tls/go-server/app/server.go [hunk 6]
```diff
@@ -0,0 +1,80 @@
+package main
```
Comment (beiwei30) [inline tls/go-server/app/server.go:1]: pls. add apache license header.

File: tls/go-server/app/start.sh [hunk 7]
```diff
@@ -0,0 +1,4 @@
+export CONF_PROVIDER_FILE_PATH="../conf/server.yml"
```
Comment (beiwei30) [inline tls/go-server/app/start.sh:1]: avoid use shell script to start. try if build/makefile work or not. pls. refer to HOWTO.md

File: tls/go-server/app/user.go [hunk 8]
```diff
@@ -0,0 +1,47 @@
+package main
```
Comment (beiwei30) [inline tls/go-server/app/user.go:1]: pls. add apache license header.

File: tls/go-server/app/version.go [hunk 9]
```diff
@@ -0,0 +1,5 @@
+package main
```
Comment (beiwei30) [inline tls/go-server/app/version.go:1]: remove it.

**Review by beiwei30 — CHANGES_REQUESTED** [review]

File: tls/go-client/cmd/client.go [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+	"path/filepath"
+	"syscall"
+	"time"
+)
+
+import (
+	getty "github.com/apache/dubbo-getty"
+	hessian "github.com/apache/dubbo-go-hessian2"
+	gxlog "github.com/dubbogo/gost/log"
+	"github.com/apache/dubbo-go-samples/tls/go-client/pkg"
```
Comment (beiwei30) [inline tls/go-client/cmd/client.go:34]: put "tls/go-client/pkg" into separated import group, and combine others into the group below.

File: tls/go-client/conf/client.yml [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+    protocol: "zookeeper"
+    timeout	: "3s"
+    address: "127.0.0.1:2181"
+    usern ame: ""
+    password: ""
+
+
+references:
+  "UserProvider":
+    # 可以指定多个registry，使用逗号隔开;不指定默认向所有注册中心注册
```
Comment (beiwei30) [inline tls/go-client/conf/client.yml:30]: remove this line of comment.

File: tls/go-client/pkg/user.go [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+ */
+
+package pkg
+
+import (
+	"context"
+	"time"
+)
+
+type User struct {
```
Comment (beiwei30) [inline tls/go-client/pkg/user.go:25]: pls. reformat the code.

File: tls/go-server/cmd/server.go [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+	"time"
+)
+
+import (
+	getty "github.com/apache/dubbo-getty"
+	"github.com/apache/dubbo-go-samples/tls/go-server/pkg"
```
Comment (beiwei30) [inline tls/go-server/cmd/server.go:31]: again, pls. regroup the import. Move 'tls/go-server/pkg' into separated package.

File: tls/go-server/cmd/server.go [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+	_ "github.com/apache/dubbo-go/protocol/dubbo"
+	_ "github.com/apache/dubbo-go/registry/protocol"
+	_ "github.com/apache/dubbo-go/common/proxy/proxy_factory"
+	_ "github.com/apache/dubbo-go/filter/filter_impl"
+	_ "github.com/apache/dubbo-go/cluster/cluster_impl"
+	_ "github.com/apache/dubbo-go/cluster/loadbalance"
+	_ "github.com/apache/dubbo-go/registry/zookeeper"
+)
+
+// 生存时间
```
Comment (beiwei30) [inline tls/go-server/cmd/server.go:47]: use English for comments.

File: tls/go-server/cmd/server.go [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+	caPemPath, _ := filepath.Abs("../certs/ca.pem")
+	config.SetSslEnabled(true)
+	config.SetServerTlsConfigBuilder(&getty.ServerTlsConfigBuilder{
+		ServerKeyCertChainPath:        serverPemPath,
+		ServerPrivateKeyPath:          serverKeyPath,
+		ServerTrustCertCollectionPath: caPemPath,
+	})
+}
+
+/*
```
Comment (beiwei30) [inline tls/go-server/cmd/server.go:64]: use English for comments.

File: tls/go-server/cmd/server.go [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+	需要配置环境变量
+	export CONF_PROVIDER_FILE_PATH="xx"
+	export APP_LOG_CONF_FILE="xx"
+ */
+
+func main() {
+	// 在运行的时候序列化
```
Comment (beiwei30) [inline tls/go-server/cmd/server.go:71]: use English for comments.

File: tls/go-server/cmd/server.go [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+	hessian.RegisterPOJO(&pkg.User{})
+	// 加载配置
```
Comment (beiwei30) [inline tls/go-server/cmd/server.go:73]: use English for comments.

File: tls/go-server/cmd/server.go [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+	config.Load()
+	// 优雅的结束程序
```
Comment (beiwei30) [inline tls/go-server/cmd/server.go:75]: use English for comments.

File: tls/go-server/cmd/server.go [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+	initSignal()
+}
+
+// 优雅的结束程序
```
Comment (beiwei30) [inline tls/go-server/cmd/server.go:79]: use English for comments.

File: tls/go-server/conf/server.yml [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+
+registries :
+  "demoZk":
+    protocol: "zookeeper"
+    timeout	: "3s"
+    address: "127.0.0.1:2181"
+
+services:
+  "UserProvider":
+    # 可以指定多个registry，使用逗号隔开;不指定默认向所有注册中心注册
```
Comment (beiwei30) [inline tls/go-server/conf/server.yml:21]: remote this line of comment

File: tls/go-server/conf/server.yml [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+    registry: "demoZk"
+    protocol : "dubbo"
+    # 相当于dubbo.xml中的interface
```
Comment (beiwei30) [inline tls/go-server/conf/server.yml:24]: remove this line.

**Review by beiwei30 — CHANGES_REQUESTED** [review]

File: tls/go-server/conf/server.yml (hunk 21 above)
Comment (cjphaha) [reply tls/go-server/conf/server.yml:24]: > remove this line.
Thank you for reviewing the code for me. I'll change it later！

File: tls/README.md [hunk 22]
```diff
@@ -0,0 +1,61 @@
+## TLS encrypted transmission
+
+The TLS encrypted transmission of Dubbo go is realized by Dubbo Getty https://github.com/apache/dubbo-getty . Before using TLS, you need to be prepared as follows.
```
Comment (AlexStocks) [inline tls/README.md:3]: dubbo-go. dubbo-getty.

File: tls/README_zh.md [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+	config.SetSslEnabled(true)
+  // 导入tls配置
+	config.SetClientTlsConfigBuilder(&getty.ClientTlsConfigBuilder{
+		ClientPrivateKeyPath:          clientKeyPath,
+		ClientTrustCertCollectionPath: caPemPath,
+	})
+}
+```
+
+其余设置同helloworld保持一致皆可。
```
Comment (AlexStocks) [inline tls/README_zh.md:63]: 补充下 hello world 项目的连接

File: tls/go-client/conf/client.yml (hunk 11 above)
Comment (AlexStocks) [reply tls/go-client/conf/client.yml:30]: why remove this line? maybe we can translate it into english.

File: tls/go-server/cmd/server.go (hunk 19 above)
Comment (cjphaha) [reply tls/go-server/cmd/server.go]: I have modified the problems in PR, thank you very much!

File: tls/README_zh.md (hunk 23 above)
Comment (cjphaha) [reply tls/README_zh.md]: I have modified it in the commit later, sorry for the trouble！

File: tls/go-server/pkg/user.go [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+
+type User struct {
+	Id   string
+	Name string
+	Age  int32
+	Time time.Time
+}
+
+type UserProvider struct {
+
```
Comment (AlexStocks) [inline tls/go-server/pkg/user.go:48]: pls delete this blank line

File: tls/go-server/pkg/user.go [hunk 25]
```diff
@@ -0,0 +1,63 @@
...
+)
+
+func init() {
+	config.SetProviderService(new(UserProvider))
+	// ------for hessian2------
+	hessian.RegisterPOJO(&User{})
+}
+
+type User struct {
+	Id   string
```
Comment (AlexStocks) [inline tls/go-server/pkg/user.go:41]: pls change the Id to ID in this sample.

**Review by AlexStocks — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Review by zhaoyunxing92 — APPROVED** [review]
approved these changes

**Review by beiwei30 — APPROVED** [review]

**Pull request merged**